File: Sources/Ably/ARTLogAdapter.m
#import "ARTLogAdapter.h"
#import "ARTLogAdapter+Testing.h"
#import "ARTLog.h"

@implementation ARTLogAdapter

- (instancetype)initWithLogger:(ARTLog *)logger {
    if (self = [super init]) {
        _logger = logger;
    }

    return self;
}

- (void)log:(NSString *)message withLevel:(ARTLogLevel)level file:(NSString *)fileName line:(NSInteger)line {
    NSString *const augmentedMessage = [NSString stringWithFormat:@"(%@:%ld) %@", fileName, (long)line, message];
    [self.logger log:augmentedMessage withLevel:level];
}

- (ARTLogLevel)logLevel {
    return self.logger.logLevel;
}

- (void)setLogLevel:(ARTLogLevel)logLevel {
    self.logger.logLevel = logLevel;
}

@end

File: Sources/AblySwift/ARTLogAdapter.swift
import Foundation

// swift-migration: original location ARTLogAdapter.h, line 15 and ARTLogAdapter.m, line 5
internal class ARTLogAdapter: NSObject, ARTVersion2Log {
    
    // swift-migration: original location ARTLogAdapter+Testing.h, line 9 and ARTLogAdapter.m, line 9
    internal let logger: ARTLog
    
    // swift-migration: original location ARTLogAdapter.h, line 17
    private override init() {
        fatalError("init() is not available")
    }
    
    // swift-migration: original location ARTLogAdapter.h, line 25 and ARTLogAdapter.m, line 7
    internal init(logger: ARTLog) {
        self.logger = logger
        super.init()
    }
    
    // swift-migration: original location ARTLogAdapter.m, line 15
    func log(_ message: String, withLevel level: ARTLogLevel, file fileName: String, line: Int) {
        let augmentedMessage = "(\(fileName):\(line)) \(message)"
        logger.log(augmentedMessage, withLevel: level)
    }
    
    // swift-migration: original location ARTLogAdapter.h, line 30 and ARTLogAdapter.m, line 20
    var logLevel: ARTLogLevel {
        get {
            return logger.logLevel
        }
        set {
            logger.logLevel = newValue
        }
    }
}